Suggest changes to beta phase (#20280)
nvaccess/nvda: projectDocs/community/releaseProcess.md

### Beta phase

* A commit will be selected from the `master` branch and merged into `beta`.

> * Once a beta commit is chosen, no further changes are included in the release, except the following will be considered:

Reviewer: ```suggestion
* Once a beta commit is chosen, no further changes except the following are included in the release:
```

or

```suggestion
* Once a beta commit is chosen, only the following types of changes are included in the release:
```

  * addressing a bug introduced in the current or previous release cycle.

>   * addressing a `p1` filed in the current or previous release cycle.

Reviewer: How does this differ from the bug case above? If a P1 is filed that affects the last 10 versions, is it still considered for inclusion in the betas?

Author: Yes, for example an OS/browser/critical software regression that needs an urgent fix. it may affect all versions of NVDA because they broke it, but we may still need to fix it. I think with p1s, and potentially p2s, that when they get filed we should be pushing out fixes for them ASAP. However, if a p1/p2 can't be fixed immediately, or stagnates for some time, we can probably delay it a bit further.

>   * reverting code introduced in the current release cycle.

Reviewer: I think we should include a carve-out for documentation updates. We could limit this to minor docs changes if we want to limit the burden on translators.

```suggestion
  * reverting code introduced in the current release cycle.
  * updates to documentation introduced in this release cycle.
```

or

```suggestion
  * reverting code introduced in the current release cycle.
  * Updates to address inaccuracies or miner spelling/grammar/formatting mistakes in documentation introduced in this release cycle.
```

### Beta phase

> * A commit without any known serious issues, will be selected from the 'master' branch and merged into 'beta'.

Author: ```suggestion
* A commit without any known serious issues, will be selected from the `master` branch and merged into `beta`.
```

> * A commit without any known serious issues, will be selected from the `master` branch and merged into `beta`.

Author: ```suggestion
* A commit will be selected from the `master` branch and merged into `beta`.
```

* Once a beta commit is chosen, only the following types of changes are included in the release:
  * addressing a bug introduced in the current or previous release cycle.

>   * addressing a `p1` filed in the current or previous release cycle.

Author: Including p2s because of #20319 as an example

```suggestion
  * addressing a `p1` or `p2` filed in the current or previous release cycle.
```

  * reverting code introduced in the current release cycle.
* Documentation changes will be reviewed.
A release summary will be added to the change log for the beta.

> * A tagged "beta release" will be created for wider testing.

Author: ```suggestion
* Once considered stable enough, a tagged "beta release" will be created for wider testing.
```

>   * updates to documentation introduced in this release cycle.

Author: @SaschaCowley - thoughts on this instead? Covers a situation like #20192 better

```suggestion
  * updates to documentation introduced in the current or previous release cycle.
```

>   * updates to documentation introduced in this release cycle.

Author: ```suggestion
  * updates to documentation introduced in this release cycle.
  * updates to translations
```

>   * updates to documentation introduced in the current or previous release cycle.

Author: ```suggestion
  * updates to documentation introduced in the current or previous release cycle.
  * updates to translations.
```
